### ISSUE TITLE
Upload large mp4 finished with error message

### ISSUE THREAD

**Issue by ksapp-dell** [nextcloud/android#6405]
### Steps to reproduce
1. Upload an mp4 file bigger than 5
600 Mb - 
2. 
3. 

### Expected behaviour
- File upload without Error

### Actual behaviour
- the file upload finishes with error message ,Hochladen fehlgeschlagen - Fehler beim warten auf den Server'. But the file is uploaded completly to the server.

Upload via Browser from the same phone works fine. Therefore I assume that the server is configured correctly.

### Environment data
Android version:
8.0
Device model: Samsung S7

Stock or customized system:
Stock
Nextcloud app version:
3.12.0

Also the dev Version shows the same behaviour. But the log shows something with 'Timeout'

Nextcloud server version:
18.06
Reverse proxy:
?
#### Web server error log
```

```

#### Nextcloud log (data/nextcloud.log)
Nothing to show. No errors in the Nextcloud  log.
```nsert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Can you provide us additional infos via logcat? https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by ksapp-dell** [comment]
Hi, i have the logcat file downloaded from the phone. 

I removed some staff which seems private for me. I hope it helps.

Oliver

[logcatOutput2.txt](https://github.com/nextcloud/android/files/4868290/logcatOutput2.txt)

**Comment by tobiasKaminsky** [comment]
After uploading all the chunks they get assembled:
REQUEST MOVE /nextcloud/remote.php/dav/uploads/xxxxxxx/c40051fe6e41ad12296145550c032ee1/.file

And this seems to take too long.

@nextcloud/server-triage how long shall we wait on this assembling of chunks?

**Comment by ksapp-dell** [comment]
Hi, but it seems not to have influence on the file. The file works and is not broken. It looks just like an error message with not really a problem begins it.

**Comment by tobiasKaminsky** [comment]
@nextcloud/server-triage do you have an explanation for this? I would have assumed that then the file will not get assembled/moved to correct place?

**Comment by SirSkorpan** [comment]
I seem to have the same issue, when a file takes too long to assemble the android app (version 3.12.0) reports `Connection error`, under the Uploads tab after reaching 100% and waiting for a short time. 

On the server side everything seems to be fine (i.e it's assembled and moved to the expected directory). I expect #5609 refers to the same problem. 

In my case I'm using a RPi 4 as the server (Nextcloud 18.0.4) and the latest example was a 1.7G file that took about 2min to assemble, the android app reported the error after 20s.

**Comment by ksapp-dell** [comment]
Anything  new here ? The issue mentioned above Senn the same, but thanks to the bot, the issue was closed without any solution.
May be the ,time out' should be adjusted (-;

**Comment by rullzer** [comment]
There is not much we can do here. If it takes to long the webserver or the client will most likely kill the connection. resulting in an error. However in the background assembling will continue so after a while the file will be properly uploaded.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky the only way to "fix" this on the client side imho would be to introduce another upload status which we set based on the timeout event like "check successful status" or "pending check" and then kind of "retry" these uploads as in check if the file is available on the server (like an hour later) and if so set the status to "successful/uploaded".

What do you think?

**Comment by SirSkorpan** [comment]
I would say that the main problem is that the app continues to try to upload the large file after the file has been assembled. Meaning, after 2-3 minutes (in my case) the file is actually present on the server but the client tries to upload over and over (using bandwidth and data). At some point you may be asked to resolve a conflict, but choosing to keep the one on the server doesn't actually remove the upload task from the list.

**Comment by ksapp-dell** [comment]
Ping. To prevent the bot from closing this again (-;

**Comment by tobiasKaminsky** [comment]
> another upload status which we set based on the timeout event like "check successful status" or "pending check"

As there can also be other timeouts, this is something server should send us.

**Comment by SirSkorpan** [comment]
I agree that if the app was to automatically act on timeouts the server would have to provide information about the reason for the timeout.

One suggestion could be to have a "large file"-mode which, if turned on, would wait X amount of time (say 5min as default) before reporting an error for the upload. Maybe the user could choose what a "large file" means, default could be about 0.5GB. After an upload of a "large file" times out maybe the app waits X min and then checks with the server. If the file is on the server it marks the upload as complete.

(I'm not sure how nextcloud determines if the file it attempted to upload and the file already on the server is different, maybe it would need to show a conflict message)

At the moment I do get a conflict message after a while, however I get duplicate messages and they don't disappear when I choose to keep the "Already existing file". I'm not sure if this is due to the same issue or if it's related to #6510 
![Screenshot_20200724-172511_Nextcloud](https://user-images.githubusercontent.com/22483583/88410155-44d92b80-cdd6-11ea-8d7e-9d068db89718.jpg)

**Comment by ksapp-dell** [comment]
Isn't it possible, that the server sends a kind of ,I am still working in it ' signal like every 5 seconds to the app to inform the app, that the server is not ready I stead if waiting a fixed time period before reporting time out ?
However- who initiates the time out at the moment- the app ?

**Comment by iwismer** [comment]
I think I have the same issue as well.
I use auto upload for my signal messages backup, which is about 2gb. The file uploads successfully (as in the file is intact of nextcloud), but the app usually reports an error, and then tries to reupload the file. The file is continuously reuploaded until I manually cancel it (or nextcloud doesn't report an error), which drains a lot of battery. This happens every day when signal generates its backup file.

**Comment by Grunthos** [comment]
Seeing this also. Is it possible to set the timeout based on file size? Or make it a user setting?

**Comment by ppascher** [comment]
Having the same issue. I like @ksapp-dell 's idea of sending a signal to keep the connection up. Would that be feasible with the current server/app interaction? Where is the timeout set currently? Is this i.e. the nginx setting? Some php parameter?

**Comment by funkypopcorn** [comment]
Anything new here?

The error seems to still exists after one year.
And as smartphones become more and more powerfull (4-5.2k video recording) files, especially video files tend to get bigger and bigger.

**Comment by kriztan** [comment]
Same here with server version 21.0.4 on debian and Android app version 3.17.0. It's annoying and disruptive.
